magefile.go

```diff
@@ -55,3 +56,74 @@ func (Lint) Codegen(ctx context.Context) error {
 
 	return nil
 }
+
+type Build mg.Namespace
+
+// Dagger will build the dagger binary
+func (Build) Dagger(ctx context.Context) error {
+	c, err := dagger.Connect(ctx)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	builder := c.Container().From("golang:1.18.2-alpine")
+
+	workdir := c.Host().Workdir()
+	goMod, err := workdir.Read().File("go.mod").ID(ctx)
+	if err != nil {
+		return err
+	}
+
+	goSum, err := workdir.Read().File("go.sum").ID(ctx)
+	if err != nil {
+		return err
+	}
+
+	builder = builder.Exec(dagger.ContainerExecOpts{
+		Args: []string{"mkdir", "/app"},
+	})
+
+	fs := builder.FS()
+	fs = fs.WithCopiedFile("/app/go.mod", goMod)
+	fs = fs.WithCopiedFile("/app/go.sum", goSum)
+
+	modFSID, err := fs.ID(ctx)
+	if err != nil {
+		return err
+	}
+
+	builder = builder.WithFS(modFSID).WithWorkdir("/app")
+	builder = builder.Exec(dagger.ContainerExecOpts{
+		Args: []string{"go", "mod", "download"},
+	})
+
+	src, err := workdir.Read().ID(ctx)
+	if err != nil {
+		return err
+	}
+
+	builder = builder.WithMountedDirectory("/app", src).WithWorkdir("/app")
+
+	builder = builder.Exec(dagger.ContainerExecOpts{
+		Args: []string{"mkdir", "/app/build"},
+	})
+
+	builder = builder.Exec(dagger.ContainerExecOpts{
+		Args: []string{"go", "build", "-o", "/app/build/dagger", "/app/cmd/dagger"},
```

Comment:
We should set `CGO_ENABLED=0`, otherwise we'll probably be linked against musl due to importing `net` (if not now in the future)